code/modules/mob/living/slime/feeding.dm

```diff
@@ -0,0 +1,119 @@
+/mob/living/slime/proc/set_nutrition(var/amt)
+	nutrition = Clamp(amt, 0, initial(nutrition))
+
+/mob/living/slime/proc/check_valid_feed_target(var/mob/living/M, var/check_already_feeding = TRUE, var/silent = FALSE)
+	if(QDELETED(M) || !istype(M) || !isturf(M.loc))
+		return FEED_RESULT_INVALID
+	if(M == src)
+		if(!silent)
+			to_chat(src, SPAN_WARNING("You cannot feed on yourself."))
+		return FEED_RESULT_INVALID
+	if(check_already_feeding && feeding_on)
+		if(!silent)
+			to_chat(src, SPAN_WARNING("You are already feeding on \the [feeding_on.resolve()]."))
+		return FEED_RESULT_INVALID
+	if(!istype(M) || issilicon(M) || isslime(M))
+		if(!silent)
+			to_chat(src, SPAN_WARNING("You cannot feed on \the [M]."))
+		return FEED_RESULT_INVALID
+	if(!Adjacent(M))
+		if(!silent)
+			to_chat(src, SPAN_WARNING("\The [M] is too far away."))
+		return FEED_RESULT_INVALID
+	if(M.get_blocked_ratio(null, TOX, damage_flags = DAM_DISPERSED | DAM_BIO) >= 1)
+		if(!silent)
+			to_chat(src, SPAN_WARNING("\The [M] is protected from your feeding."))
+		return FEED_RESULT_INVALID
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		if(H.species.species_flags & (SPECIES_FLAG_NO_POISON|SPECIES_FLAG_NO_SCAN))
+			if(!silent)
+				to_chat(src, SPAN_WARNING("You cannot feed on \the [M]."))
+			return FEED_RESULT_INVALID
+	if(M.stat == DEAD)
+		if(!silent)
+			to_chat(src, SPAN_WARNING("\The [src] is dead."))
+		return FEED_RESULT_DEAD
+	if(M.getCloneLoss() >= M.maxHealth * 1.5)
+		if(!silent)
+			to_chat(src, SPAN_WARNING("\The [M] is too degraded to feed upon."))
+		return FEED_RESULT_DEAD
+	if(M.currently_being_eaten_by_a_slime(src))
+		if(!silent)
+			to_chat(src, SPAN_WARNING("Another slime is already feeding on \the [M]."))
+		return FEED_RESULT_INVALID
+	return FEED_RESULT_VALID
+
+/mob/living/slime/proc/set_feeding_on(var/mob/living/victim)
+	if(feeding_on)
+		var/mob/feed_mob = feeding_on.resolve()
+		GLOB.moved_event.unregister(src, src)
+		GLOB.moved_event.unregister(feed_mob, src)
+		GLOB.destroyed_event.unregister(feed_mob, src)
+	feeding_on = null
+	if(victim)
+		if(feeding_on == weakref(victim))
```

Comment:
weird flow here
first we unregister and null feeding on, and THEN we check if we're getting same target as one we had.
I think this check should be at the top before unregistering so you don't jiggle events back and forth needlesly.